pkg/tenant/tenants.go

```diff
@@ -0,0 +1,121 @@
+package tenant
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/rs/zerolog/log"
+)
+
+// CreateTenant creates a new tenant in the database
+func (s *Server) CreateTenant(c *gin.Context) {
+	var (
+		err    error
+		tenant *api.Tenant
+	)
+
+	if err = c.BindJSON(&tenant); err != nil {
+		log.Warn().Err(err).Msg("could not parse tenant create request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse request"))
+		return
+	}
+
+	if tenant.ID == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrTenantIDRequired))
+	}
```

Comment:
I think this should be the opposite - when we create a tenant, we don't want the user to specify an ID - we want to assign the ID ourselves. 

```suggestion
	if tenant.ID != "" {
		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrInvalidTenantField))
		return
	}
```